iocore/net/YamlSNIConfig.cc

```diff
@@ -173,7 +181,30 @@ template <> struct convert<YamlSNIConfig::Item> {
     }
 
     if (node[TS_fqdn]) {
-      item.fqdn = node[TS_fqdn].as<std::string>();
+      swoc::TextView fqdn{node[TS_fqdn].Scalar()};
+      auto port_view{fqdn.take_suffix_at(':')};
+      if (port_view && fqdn) {
+        item.fqdn = fqdn;
+        auto min{port_view.split_prefix_at('-')};
+        if (!min) {
+          min = port_view;
+        }
+        auto const &max{port_view};
+
+        swoc::TextView parsed_min;
+        long min_port{swoc::svtoi(min, &parsed_min)};
+        swoc::TextView parsed_max;
+        long max_port{swoc::svtoi(max, &parsed_max)};
+        if (parsed_min != min || min_port < 1 || parsed_max != max || max_port > 65535 || max_port < min_port) {
+          std::string out;
+          swoc::bwprint(out, "bad port range: {}-{}", min, max);
+          throw YAML::ParserException(node[TS_fqdn].Mark(), out);
+        }
+
+        item.port_ranges.emplace_back(std::make_pair(min_port, max_port));
+      } else {
+        item.fqdn = port_view;
```

Comment:
A comment explaining if there is no `:`, then `fqdn` is empty and `port_view` contains the server name.